File: src/Compare Version Numbers/Compare Version Numbers.cpp
class Solution {
	public:
		string mysplit(string version, int &pos)
		{
			string ret = "";
			for(int i = pos ; i < version.length() ; i++)
			{
				pos++;
				if(version[i] == '.')	 break;
				else ret += version[i];
			}
			return ret;
		}

		int compareVersion(string version1, string version2) 
		{
			int pos1 = 0, pos2 = 0;

			while(1)
			{
				string str1 = mysplit(version1, pos1);
				string str2 = mysplit(version2, pos2);
				int num1 = atoi(str1.c_str());
				int num2 = atoi(str2.c_str());

				if(num1 > num2) return 1;
				else if (num1 < num2) return -1;
				else
				{
					/*if end of the version string, return 0 (same)*/
					if(pos1 == version1.length() && pos2 == version2.length() ) return 0;
				}
			}
		}
};

File: src/Compare Version Numbers/Compare Version Numbers.py
class Solution:
	# @param version1, a string
	# @param version2, a string
	# @return an integer
	def compareVersion(self, version1, version2):
		group1 , group2 = version1.split('.'), version2.split('.')
		group1_size, group2_size, pos = len(group1), len(group2), 0
		for v1, v2 in zip(group1, group2):
			pos += 1
			if int(v1) > int(v2):
				return 1
			elif int(v1) < int(v2):
				return -1
			else:
				if pos == group1_size and pos == group2_size:
					return 0
				elif pos == group1_size: 
					#version1 end first
					# if 1.0 v.s. 1 , return 0
					return 0 if int(group2[pos]) == 0 else -1
				elif pos == group2_size: 
					return 0 if int(group1[pos]) == 0 else 1
